src/pip/_internal/utils/subprocess.py

```diff
@@ -184,15 +184,16 @@ def call_subprocess(
         proc = subprocess.Popen(
             # Convert HiddenText objects to the underlying str.
             reveal_command_args(cmd),
-            stdin=subprocess.PIPE,
+            stdin=subprocess.PIPE if not stdout_only else None,
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT if not stdout_only else subprocess.PIPE,
             cwd=cwd,
             env=env,
         )
-        assert proc.stdin
         assert proc.stdout
-        proc.stdin.close()
+        if not stdout_only:
+            assert proc.stdin
+            proc.stdin.close()
```

Comment:
Why not

```python
if proc.stdin:
    proc.stdin.close()
```

Reply:
@uranusjr I did something a little bit different in the last version. I don't quite understand why we need to pass a PIPE to stdin while we don't use it, but I don't dare to change it, and I decided to keep passing one in both modes.

Reply:
I think the intent is to disallow all stdin from subprocesses. Without it, the subprocess’s stdin would be inherited from pip’s, i.e. subprocesses are able to show prompts etc.

Reply:
More fun is that it waits in CI for prompts, instead dof outright failing as well.